include/my_valgrind.h

```diff
@@ -29,6 +29,8 @@
 
 #if defined(HAVE_VALGRIND) && defined(HAVE_valgrind)
 # include <valgrind/memcheck.h>
+# define MY_ASAN_POISON_MEMORY_REGION(a,len) ((void)(a), (void)(len))
+# define MY_ASAN_UNPOISON_MEMORY_REGION(a,len) ((void)(a), (void)(len))
```

Comment:
I tend to disagree with that.
Consider a case:
```c++
auto ptr = get_ptr();
auto size = get_size();
MY_ASAN_UNPOISON_MEMORY_REGION(a,len) ((void)(a), (void)(len));
```
Here `ptr` and `size` is used only for unpoisoning. In build without ASAN there could be `unused variable` warning. And `(void)var` protects from such warnings.

Reply:
I think that in that example, it would be better to avoid using named temporaries, and simply write
```c++
MY_ASAN_UNPOISON_MEMORY_REGION(get_ptr(), get_size());
```